libsql-server/src/namespace/configurator/mod.rs

```diff
@@ -73,13 +99,30 @@ impl NamespaceConfigurators {
 pub trait ConfigureNamespace {
     fn setup<'a>(
         &'a self,
-        ns_config: &'a NamespaceConfig,
         db_config: MetaStoreHandle,
         restore_option: RestoreOption,
         name: &'a NamespaceName,
         reset: ResetCb,
         resolve_attach_path: ResolveNamespacePathFn,
         store: NamespaceStore,
         broadcaster: BroadcasterHandle,
-    ) -> Pin<Box<dyn Future<Output = crate::Result<super::Namespace>> + Send + 'a>>;
+    ) -> Pin<Box<dyn Future<Output = crate::Result<Namespace>> + Send + 'a>>;
+
+    fn cleanup<'a>(
+        &'a self,
+        namespace: &'a NamespaceName,
+        db_config: &'a DatabaseConfig,
+        prune_all: bool,
+        bottomless_db_id_init: NamespaceBottomlessDbIdInit,
+    ) -> Pin<Box<dyn Future<Output = crate::Result<()>> + Send + 'a>>;
```

Comment:
why not use `async_trait`?

Reply:
no real reason, except that async_trait doesn't play nicely with LSP, and it expends to something quite trivial...